python/mxnet/base.py

```diff
@@ -26,11 +26,11 @@
 import sys
 import inspect
 import platform
-import numpy as np
+import numpy as _np
 
 from . import libinfo
 
-__all__ = ['MXNetError']
+__all__ = ['MXNetError', 'is_np_compat', 'set_np_compat', 'enable_np_compat', 'disable_np_compat']
```

Comment:
I'd suggest we have one scope (e.g. `mx.np_compat(active=True)`) instead of two.
Also, shall we use the scope to implement a decorator so that it's easier to mark the functions where the flag can be turned on and off? (e.g. `@mx.np_compat(active=True)def foo():`, or you may need another name for the decorator)

Reply:
One scope is a good suggestion.

The decorator may have to have a different name than `np_compat` since the `__init__` function would have a different signature. As of now, I don't see the strong need of such a decorator to move forward. How about we defer this till the time we really need it?

Reply:
When providing APIs of Gluon that are implemented with the numpy semantics, using a decorator would be easier to use than a scope as it doesn't require the indentation. It also means minimum amount of change for adapting to the next major version when we fully commit to numpy semantics. I'd recommend having the from the beginning.

Reply:
That was needed when we wanted to dispatch fluent methods to numpy ops. But since we have decided to make a dedicated ndarray data type for numpy ops, such decorator may not be needed. We must clear out the thoughts for such use cases before adding the api. It cannot recalled once out. Let's discuss offline.

Reply:
Done.